deployment/articles-dxp/01-saml/04-setting-up-service-provider.markdown

```diff
@@ -0,0 +1,163 @@
+    keystore.
+ 
+    **Service Provider**: This tab manages basic and advanced configurations for
+    the SP.
+ 
+    **Identity Provider Connection**: This tab manages connections to the IdP.
+    There can be multiple IdP connections.
+
+5.  You can also generate an encryption certificate. This is a separate key for
+    encrypting assertions. If you want assertions encrypted, you must generate
+    a key for this. The procedure is exactly the same as generating your
+    certificate in step 3 above. 
+
+5.  Next, you must configure an Identity Provider connection. Click on the
+    *Identity Provider Connections* tab. Enter a name for the Identity Provider,
+    enter its entity ID, and enter its metadata URL. If you have already
+    followed the previous instructions and configured a separate @product@
+    installation as an Identify provider, you'd enter the following information:
+
+    - Name: *Liferay IdP*
+    - Entity ID: [ID of IdP]
+	- Clock Skew: Set the tolerance in milliseconds between SP and IdP.
+	- Force Authn: Whether the IdP should force reauthentication regardless of
+      context.
+    - Metadata URL: http://localhost:8080/c/portal/saml/metadata (test this URL
+      first)
+	- Name Identifier Format: See settings. 
+	- Attribute Mapping: See settings. 
+	- Keep Alive URL: See settings. 
+
+    **Important**: The Liferay SAML 3.0 Provider app supports using *either*
+    a URL to a SAML IdP metadata file *or* an actual (uploaded) SAML metadata
+    XML file. The value entered in the *Metadata URL* field is persisted to the
+    database only when there a metadata URL and there is no specified metadata
+    XML file. Otherwise, @product@ keeps the original metadata URL in the
+    database. This behavior ensures that once a metadata URL has been specified,
+    there is always a metadata URL saved in the database. This way, if a portal
+    administrator forgets the previously entered metadata URL or its format, he
+    or she can look at the displayed metadata URL and choose to modify the
+    displayed metadata URL or overwrite the previously saved metadata URL by
+    specifying a metadata XML file.
+
+6.  Finally, after you save your certificate and private key information and
+    configure an Identity Provider connection, check the *Enabled* box at the top
+    of the General tab and click *Save*. @product@ is now a SAML Service Provider!
+
+Note that the SAML Service Provider session is tied to the normal session on
+the application server. Session expiration on the application server terminates
+the session on the Service Provider but does not initiate single logout. 
+
+You can add multiple IdP connections. To add another Identity Provider, click
+*Add Identity Provider* again and enter the details for the other provider. When
+users log in, they are asked to choose an identity provider, so be sure to name
+the providers so users can recognize them. 
```

Comment:
@sez11a Can we add a screenshot here? Despite the "IdP" can be different, we could show for example a setup like: IdP 1: DXP 7.1 IdP 2: DXP 7.2 + SP: DXP 7.2

Reply:
@sez11a In addition, we should also include the customization method from https://issues.liferay.com/browse/LPS-88090:

header could "Customizing the SAML Identity Profile Selector via `SamlSpIdpConnectionsProfile`"

Reply:
That would be a separate doc for developers. Created https://issues.liferay.com/browse/LRDOCS-6878